### PR TITLE
feat(skore/ComparisonReport): Add `to_markdown`

### PR DESCRIPTION
**Pull request by GaetandeCast**
Towards #2941.

<details>

<summary>Example output</summary>

# ComparisonReport: LogisticRegression, RidgeClassifier

## Estimators

- ml task: binary-classification

| report name | LogisticRegression | RidgeClassifier | 
| --- |  --- |  --- | 
| estimator name | `LogisticRegression` | `RidgeClassifier` | 
| kind | scikit-learn estimator | scikit-learn estimator | 
| fit time | 0.00299 s | 0.00297 s | 
| predict time (on test set) | 0.000836 s | 0.000445 s | 

## Metrics

```text
                        LogisticRegression  RidgeClassifier
Metric           Label                                     
Score                             0.950000         0.950000
Accuracy                          0.950000         0.950000
Precision        0                0.923077         0.923077
                 1                1.000000         1.000000
Recall           0                1.000000         1.000000
                 1                0.875000         0.875000
ROC AUC                           1.000000         1.000000
Log loss                          0.165920              NaN
Brier score                       0.043655              NaN
Fit time (s)                      0.002987         0.002967
Predict time (s)                  0.000836         0.000445

Get the metrics results and control output format with: `report.metrics.summarize().frame()`.
```

## Checks (fast mode)

```text
Checks summary: 1 issue(s), 2 tip(s), 5 passed, 0 ignored.
Issues:
- [SKD008] Highly correlated input features. Detected in: [LogisticRegression], [RidgeClassifier]. Read more about this here: https://docs.skore.probabl.ai/dev/user_guide/automated_checks.html#skd008-correlated-features.
Tips:
- [SKD006] Coefficient interpretation. Detected in: [LogisticRegression], [RidgeClassifier]. Read more about this here: https://docs.skore.probabl.ai/dev/user_guide/automated_checks.html#skd006-unscaled-coefficients.
- [SKD016] Estimator not tuned. Detected in: [LogisticRegression], [RidgeClassifier]. Read more about this here: https://docs.skore.probabl.ai/dev/user_guide/automated_checks.html#skd016-estimator-not-tuned.
Passed:
- [SKD001] Potential overfitting
- [SKD002] Potential underfitting
- [SKD004] High class imbalance
- [SKD009] Model worse than baseline
- [SKD010] Model slower than baseline
Mute a check with .checks.summarize(ignore=['<code>']).
```

Get the full checks results with: `report.checks.summarize()`.

## Data (Same dataset version)
Computed on the full dataset (n_rows=100, n_columns=6, n_constant_columns=0).

| column | dtype | null_count | n_unique |
| ------ | ----- | ---------- | -------- |
| f_0 | Float64DType | 0 | 100 |
| f_1 | Float64DType | 0 | 100 |
| f_2 | Float64DType | 0 | 100 |
| f_3 | Float64DType | 0 | 100 |
| f_4 | Float64DType | 0 | 100 |
| Target | Int64DType | 0 | 2 |

## Data (Different dataset version)
| report name | n_rows | n_columns |
| --- | --- | --- |
| LogisticRegression | 100 | 6 |
| RidgeClassifier | 100 | 3 |
</details>

### CODE REVIEW

File: skore/src/skore/_sklearn/_cross_validation/report.py [hunk 1]
```diff
@@ -721,11 +721,7 @@ def to_markdown(self) -> str:
                     f" (± {timings.loc['Predict time test (s)', 'std']:.3f})"
                 ),
                 "n_folds": len(self.reports_),
-                "splitter_repr": (
-                    repr(self.splitter)
-                    if self.splitter is not None
-                    else f"{len(self.split_indices)} folds"
-                ),
```
Comment (GaetandeCast) [inline skore/src/skore/_sklearn/_cross_validation/report.py:728]: This was too defensive.

File: skore/src/skore/_sklearn/_comparison/report.py [hunk 2]
```diff
@@ -579,7 +583,43 @@ def _get_help_legend(self) -> str:
...
+        metrics_frame = (
+            self.metrics.summarize(data_source="test")
+            .frame()
+            .rename_axis(
+                None if self._report_type == "comparison-estimator" else [None, None],
+                axis="columns",
+            )
+        )
+        if self._report_type == "comparison-cross-validation":
+            metrics_frame = metrics_frame.swaplevel(axis="columns")
```
Comment (GaetandeCast) [inline skore/src/skore/_sklearn/_comparison/report.py]: I applied the same formatting to the dataframe that is done in the html repr a few lines under to be consistent. Not very happy to have to maintain both looks if we make changes to the output of `.frame` later as suggested here https://github.com/probabl-ai/skore/pull/3001#issuecomment-4653745898 though

File: skore/src/skore/_sklearn/_cross_validation/report.py (hunk 1 above)
Comment (auguste-probabl) [reply skore/src/skore/_sklearn/_cross_validation/report.py:728]: When a CVReport is deserialized with `from_dict`, its splitter is set to `None`, so I think the if can stay

File: skore/src/skore/_utils/repr/markdown.py [hunk 3]
```diff
@@ -46,6 +47,126 @@ def report_markdown_context(report: Any) -> dict[str, object]:
     }
 
 
+def _subreport_estimator_metadata(report: Any) -> dict[str, str | None]:
```
Comment (auguste-probabl) [inline skore/src/skore/_utils/repr/markdown.py:50]: In the other PRs the numbers were formatted in the Jinja templates

File: skore/src/skore/_utils/repr/markdown.py [hunk 4]
```diff
@@ -46,6 +47,126 @@ def report_markdown_context(report: Any) -> dict[str, object]:
...
+        summary = summarize_dataframe(
+            first_report_data["dataframe"],
+            with_plots=False,
+            with_associations=False,
+            verbose=0,
+        )
+        return {
+            "data_mode": "unified",
+            **markdown_data_section(
+                summary, data_label=first_report_data["data_label"]
```
Comment (auguste-probabl) [inline skore/src/skore/_utils/repr/markdown.py:159]: Super mega nitpick: Claude says that even if all the reports have the same data, they might have different data labels (eg. if a report's estimator is pre-fitted). However, in practice this would likely be made impossible by other parts of the markdown logic. Still, the simplest may be to add an assertion that all reports have the same data label?

<details>

<summary>
Full script to reproduce the issue
</summary>

```py
"""Showcase: ComparisonReport "unified" data-mode picks the first report's data_label.

This script demonstrates the issue flagged in the PR review for
https://github.com/probabl-ai/skore/pull/3002.

Scenario
--------
In `comparison_data_markdown_context`, when every sub-report's display dataframe
is `.equals(...)` to the first one, the "unified" branch is taken and the
markdown reuses `first_report_data["data_label"]`. For estimator sub-reports,
`data_label` is `"full"` if `X_train is not None` else `"test"`.

If a comparison mixes:
  - a prefit report with no train data  -> data_label = "test"
  - a prefit report with train+test data -> data_label = "full"

and both produce the *same* display dataframe, the resulting markdown will
either claim the data is the "test" set or the "full" set, depending purely on
the *order* of the reports passed to ComparisonReport.

This script builds exactly that comparison, swaps the order, and prints the
"Data" section of the markdown for each ordering.
"""

from __future__ import annotations

import numpy as np
import pandas as pd
from sklearn.datasets import make_classification
from sklearn.linear_model import LogisticRegression

from skore import ComparisonReport, EstimatorReport
from skore._utils.repr.markdown import comparison_data_markdown_context


def render_data_section(ctx: dict) -> str:
    if ctx["data_mode"] == "unified":
        return (
            f"## Data\n"
            f"Computed on the {ctx['data_label']} dataset "
            f"(n_rows={ctx['data_n_rows']}, "
            f"n_columns={ctx['data_n_columns']}, "
            f"n_constant_columns={ctx['data_n_constant_columns']})."
        )
    return f"## Data ({ctx['data_mode']})\n{ctx['data_shapes']}"


def main() -> None:
    X_arr, y_arr = make_classification(n_samples=60, n_features=4, random_state=0)
    # Use named columns so the display dataframe is deterministic.
    X = pd.DataFrame(X_arr, columns=[f"f{i}" for i in range(X_arr.shape[1])])
    y = pd.Series(y_arr, name="target")

    estimator = LogisticRegression().fit(X, y)

    # Report A: prefit, no train data. Its display dataframe is X + y.
    # `report.X_train is None` -> data_label = "test".
    report_no_train = EstimatorReport(
        estimator,
        fit=False,
        X_test=X,
        y_test=y,
    )

    # Report B: prefit, with an *empty* train set and the same test set as A.
    # `data_source="both"` concatenates (empty) train + test, which equals X + y.
    # `report.X_train is not None` (it's an empty frame, not None) -> data_label = "full".
    # The empty y_train means `y_test` still matches Report A's `y_test`, so the
    # `_validate_reports` y_test-hash check passes.
    empty_X = X.iloc[0:0]
    empty_y = y.iloc[0:0]
    report_with_train = EstimatorReport(
        estimator,
        fit=False,
        X_train=empty_X,
        y_train=empty_y,
        X_test=X,
        y_test=y,
    )

    # Sanity check: confirm the two display dataframes really do compare equal.
    df_no_train = report_no_train.data._prepare_dataframe_for_display(
        data_source="test",
        with_y=True,
        subsample=None,
        subsample_strategy="head",
        seed=None,
    )
    df_with_train = report_with_train.data._prepare_dataframe_for_display(
        data_source="both",
        with_y=True,
        subsample=None,
        subsample_strategy="head",
        seed=None,
    )
    assert df_no_train.equals(df_with_train), (
        "Precondition failed: the two display dataframes are not equal, so the "
        "'unified' branch would not be taken."
    )
    assert np.array_equal(df_no_train.values, df_with_train.values)

    # Order 1: no-train report first -> first.data_label == "test"
    comparison_1 = ComparisonReport(
        {"prefit_no_train": report_no_train, "prefit_with_train": report_with_train}
    )
    # Order 2: with-train report first -> first.data_label == "full"
    comparison_2 = ComparisonReport(
        {"prefit_with_train": report_with_train, "prefit_no_train": report_no_train}
    )

    # Call the helper that builds the "Data" section context. This is what
    # `ComparisonReport.to_markdown` uses internally. (We bypass the full
    # `to_markdown()` call only because empty train data makes the unrelated
    # `checks.summarize()` step blow up on this contrived input -- that's
    # incidental to the issue being demonstrated.)
    ctx_1 = comparison_data_markdown_context(comparison_1)
    ctx_2 = comparison_data_markdown_context(comparison_2)

    print("=" * 72)
    print("Order 1: prefit_no_train first  (its data_label is 'test')")
    print("=" * 72)
    print(render_data_section(ctx_1))
    print()
    print("=" * 72)
    print("Order 2: prefit_with_train first  (its data_label is 'full')")
    print("=" * 72)
    print(render_data_section(ctx_2))
    print()

    print(f"data_mode in order 1: {ctx_1['data_mode']!r}")
    print(f"data_mode in order 2: {ctx_2['data_mode']!r}")
    print(f"data_label in order 1: {ctx_1['data_label']!r}")
    print(f"data_label in order 2: {ctx_2['data_label']!r}")
    assert ctx_1["data_mode"] == "unified"
    assert ctx_2["data_mode"] == "unified"
    assert ctx_1["data_label"] == "test"
    assert ctx_2["data_label"] == "full"
    print()
    print(
        "Same underlying data, same comparison -- but the markdown's 'Data'\n"
        "section labels the dataset differently depending on which report\n"
        "happens to be listed first."
    )
```

</details>

**Review by auguste-probabl — COMMENTED** [review]
Some suggestions

File: skore/src/skore/_sklearn/_comparison/report.py (hunk 2 above)
Comment (auguste-probabl) [reply skore/src/skore/_sklearn/_comparison/report.py]: refactor to a private method? maybe this logic can be in _MetricsAccessor? I don't know what we're planning for _MetricsAccessor.__repr__ but maybe it will use it too

Comment (GaetandeCast) [reply skore/src/skore/_sklearn/_comparison/report.py]: We show the default view of `summarize().frame()` since #2998. I like the idea of having a private method on the accessor.

File: skore/src/skore/_utils/repr/markdown.py (hunk 3 above)
Comment (GaetandeCast) [reply skore/src/skore/_utils/repr/markdown.py:50]: Yes but in this case there is extra logic regarding the report type. I think it is better kept in the python code.

File: skore/src/skore/_utils/repr/markdown.py (hunk 4 above)
Comment (GaetandeCast) [reply skore/src/skore/_utils/repr/markdown.py:159]: Holy nitpick. Added `and report_data["data_label"] == first_report_data["data_label"]
` in the for loop

File: skore/src/skore/_sklearn/_cross_validation/report.py (hunk 1 above)
Comment (GaetandeCast) [reply skore/src/skore/_sklearn/_cross_validation/report.py:728]: added back

**Review by auguste-probabl — APPROVED** [review]
